### PR TITLE
Convert user-profile screen to Compose

### PR DESCRIPTION
**Pull request by VanyaHuamanBottleRocket**

### CODE REVIEW

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/profile/ProfileFragment.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.bottlerocketstudios.brarchitecture.ui.profile
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.bottlerocketstudios.brarchitecture.ui.BaseFragment
+import com.bottlerocketstudios.compose.profile.ProfileScreen
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class ProfileFragment : BaseFragment<ProfileViewModel>() {
```
Comment (jaredstockton) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/profile/ProfileFragment.kt:10]: @VanyaHuamanBottleRocket Thanks for the refactor to profile! This naming reflects the intent of the UI and class much more clearly :D

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/profile/ProfileViewModel.kt [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
 import com.bottlerocketstudios.brarchitecture.data.repository.BitbucketRepository
 import com.bottlerocketstudios.brarchitecture.navigation.ExternalNavigationEvent
 import com.bottlerocketstudios.brarchitecture.navigation.NavigationEvent
 import com.bottlerocketstudios.brarchitecture.ui.BaseViewModel
+import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 
-class UserFragmentViewModel(val repo: BitbucketRepository) : BaseViewModel() {
-
-    // TODO: Remove conversion from Flow to LiveData once it is in the next stable 7.x AGP: https://developer.android.com/topic/libraries/data-binding/observability#stateflow
```
Comment (jaredstockton) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/profile/ProfileViewModel.kt]: @VanyaHuamanBottleRocket Appreciate that you've addressed this TODO and removed the comment 😄

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/profile/ProfileViewModel.kt [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
-    val avatarUrl = repo.user.map { it?.convertToUser()?.avatarUrl.orEmpty() }.asLiveData()
-    val displayName = repo.user.map { it?.displayName.orEmpty() }.asLiveData()
-    val nickname = repo.user.map { it?.nickname.orEmpty() }.asLiveData()
+class ProfileViewModel(val repo: BitbucketRepository) : BaseViewModel() {
+    val avatarUrl: Flow<String> = repo.user.map { it?.convertToUser()?.avatarUrl.orEmpty() }
+    val displayName: Flow<String> = repo.user.map { it?.displayName.orEmpty() }
+    val nickname: Flow<String> = repo.user.map { it?.nickname.orEmpty() }
+    private val bitBucketSettingUrl: String = "https://bitbucket.org/account/settings/"
```
Comment (jaredstockton) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/profile/ProfileViewModel.kt]: @VanyaHuamanBottleRocket nit: Couple of things:

1. Move to a `companion object` block at the bottom of the class
2. Make it `const`
3. Use `SCREAMING_SNAKE_CASE`

File: compose/src/main/java/com/bottlerocketstudios/compose/profile/Profile.kt [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import com.bottlerocketstudios.compose.resources.normal
+import com.bottlerocketstudios.compose.resources.tertiary
+import com.bottlerocketstudios.compose.util.Preview
+
+data class ProfileScreenState(
+    val avatarUrl: State<String>,
+    val displayName: State<String>,
+    val nickname: State<String>,
+    val onEditClicked: () -> Unit,
+    val onLogoutClicked: () -> Unit,
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/profile/Profile.kt:37]: @VanyaHuamanBottleRocket Thanks for the trailing comma 👍🏻

File: compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfilePreviewData.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.bottlerocketstudios.compose.profile
+
+import androidx.compose.runtime.mutableStateOf
+
+val ProfileMockData = ProfileScreenState (
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfilePreviewData.kt]: @VanyaHuamanBottleRocket nit: Mark as `internal` and use camelCase (lowercase the first character)

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 6]
```diff
@@ -169,6 +169,13 @@ private object Libraries {
     // Integration with observables
     const val COMPOSE_LIVE_DATA = "androidx.compose.runtime:runtime-livedata:$COMPOSE_VERSION"
 
+    // Coil
+    // https://coil-kt.github.io/coil/
+    // https://coil-kt.github.io/coil/changelog/#full-release-notes
+    private const val COIL_VERSION = "2.0.0-rc02"
```
Comment (jaredstockton) [inline buildSrc/src/main/kotlin/Dependencies.kt:175]: @VanyaHuamanBottleRocket nit: Leave a TODO to remove Glide (around https://github.com/BottleRocketStudios/Android-ArchitectureDemo/pull/54/files#diff-7d31a18676230802e0e0c2f88ad350eae9a33404b54248dce6ed161312bf76cdR252-R255) as it won't be used anymore once the Compose migration is complete

File: compose/src/main/java/com/bottlerocketstudios/compose/resources/Dimens.kt [hunk 7]
```diff
@@ -73,6 +77,8 @@ val sw360Dimensions = Dimensions(
     grid_5 = 40.dp,
     grid_5_5 = 44.dp,
     grid_6 = 48.dp,
+    grid_12_5 = 96.dp,
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/resources/Dimens.kt]: @VanyaHuamanBottleRocket Shouldn't this be `100` (12.5*8)?

**Review by jaredstockton — CHANGES_REQUESTED** [review]
@VanyaHuaman Nice work! Let me know once you've addressed the comments and I'll take another pass :)

File: compose/src/main/java/com/bottlerocketstudios/compose/resources/Dimens.kt (hunk 7 above)
Comment (VanyaHuamanBottleRocket) [reply compose/src/main/java/com/bottlerocketstudios/compose/resources/Dimens.kt]: oops. lol changing

File: compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfilePreviewData.kt (hunk 5 above)
Comment (VanyaHuamanBottleRocket) [reply compose/src/main/java/com/bottlerocketstudios/compose/profile/ProfilePreviewData.kt]: updating

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/profile/ProfileViewModel.kt (hunk 3 above)
Comment (VanyaHuamanBottleRocket) [reply app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/profile/ProfileViewModel.kt]: Updating

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 6 above)
Comment (VanyaHuamanBottleRocket) [reply buildSrc/src/main/kotlin/Dependencies.kt:175]: Added notes to the TO DO page and added a TODO above the glide dependency

**Review by jaredstockton — APPROVED** [review]

**Review by BottleRocket-Colin — APPROVED** [review]
Approved pending any changes made during our Zoom call

**Pull request merged**